dive_core/command_hierarchy.h

```diff
@@ -83,9 +84,22 @@ class Topology
     // Children info
     uint64_t GetNumChildren(uint64_t node_index) const;
     uint64_t GetChildNodeIndex(uint64_t node_index, uint64_t child_index) const;
+    uint64_t GetNextNodeIndex(uint64_t node_index) const;
+
+    // Used to get list of shared children for a top level root node
+    // All children of this top level node share this alternate set of children
+    // Each non-top-level node has begin/end indices towards this set of shared children
     uint64_t GetNumSharedChildren(uint64_t node_index) const;
     uint64_t GetSharedChildNodeIndex(uint64_t node_index, uint64_t child_index) const;
-    uint64_t GetNextNodeIndex(uint64_t node_index) const;
+
+    // Indicate the being & end index of the shared children in the top-level node that belong to
```

Comment:
just to confirm, the shared children are the pm4 packets in the command view tab, and the nodes are the drawcalls/dispatch/blit in the tree view, right? and the Start and End here indicates the start and end index of the pm4 packets of a selected drawcall/dispatch/blit, correct?

Reply:
Pretty much. Shared nodes in principle are any nodes that can have multiple parents. In this case, PM4 packets can have as parents IB nodes, event nodes, etc. In our command hierarchy, *only* PM4 packets (not their fields) are shared nodes.

The non-shared nodes are everything else, including the children of shared nodes (e.g. the "fields" for each of the PM4 packets)

> and the Start and End here indicates the start and end index of the pm4 packets of a selected drawcall/dispatch/blit, correct?

Yes, the start and end index of the shared children that are part of selected drawcalls/dispatches/blits/ibs.